scenarios/kubernetes_build.py

```diff
@@ -30,43 +30,61 @@ def check(*cmd):
     print >>sys.stderr, 'Run:', cmd
     subprocess.check_call(cmd)
 
+def update_env(env):
+    for key, value in env.items():
+        os.environ[key] = value
 
-def main(args):
-    """Build kubernetes."""
-    env = {}
-    if args.kops:
-        env['GCS_LOCATION'] = args.kops
-    if args.unstable:
-        env['DEB_CHANNEL'] = 'unstable'
+def k8s_build_and_push(args):
+    """Build and push using the standard kubernetes build/release scripts.
+
+    This is a python port of the kubernetes/hack/jenkins/build.sh script.
+    """
+    env = {
+        # Skip gcloud update checking; do we still need this?
+        'CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK': 'true',
+        # Don't run any unit/integration tests when building
+        'KUBE_RELEASE_RUN_TESTS': 'n',
+    }
+    push_build_args = ['--nomock', '--verbose', '--ci']
+    if args.release:
+        push_build_args.append('--bucket=%s' % args.release)
     if args.suffix:
-        env['KUBE_GCS_RELEASE_SUFFIX'] = args.suffix
-    env['KUBE_FASTBUILD'] = 'true' if args.fast else 'false'
+        push_build_args.append('--gcs-suffix=%s' % args.suffix)
     if args.federation:
+        # TODO: do we need to set these?
         env['PROJECT'] = args.federation
-        env['FEDERATION'] = 'true'
         env['FEDERATION_PUSH_REPO_BASE'] = 'gcr.io/%s' % args.federation
-    if args.release:
-        env['KUBE_GCS_RELEASE_BUCKET'] = args.release
-        env['KUBE_GCS_RELEASE_BUCKET_MIRROR'] = args.release
-
+        push_build_args.append('--federation')
+    if args.registry:
+        push_build_args.append('--docker-registry=%s' % args.registry)
+    update_env(env)
 
-    for key, value in env.items():
-        os.environ[key] = value
-
-    if args.script == 'make gcs-publish-ci':
-        # TODO(fejta): fix this hack
-        check('make', 'gcs-publish-ci')
+    check('make', 'clean')
+    if args.fast:
+        check('make', 'quick-release')
     else:
-        check(args.script)
+        check('make', 'release')
+    check('../release/push-build.sh', *push_build_args)
 
+def main(args):
+    """Build and push."""
+    if not args.script:
+        k8s_build_and_push(args)
+        return
+    env = {}
+    if args.kops:
+        env['GCS_LOCATION'] = args.kops
+    if args.unstable:
+        env['DEB_CHANNEL'] = 'unstable'
+    update_env(env)
+
+    check(args.script.split(' '))
```

Comment:
I think you need `check(*args.script.split(' '))`, and seems for now `check(args.script)` is enough